src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -53,6 +54,13 @@ struct LinkCollector<'a, 'tcx: 'a, 'rcx: 'a> {
     is_nightly_build: bool,
 }
 
+#[derive(Debug, Copy, Clone)]
+enum Namespace {
```

Comment:
You should be able to reuse `rustc::hir::def::Namespace` here.

Reply:
`PathKind` is also actually `Option<Namespace>`.

Reply:
And `is_val: bool` in this file should preferably replaced with `Namespace` as well if possible.